pages/dashboard/events.tsx

```diff
@@ -1,33 +1,111 @@
+import React, { useState } from "react";
 import { GetServerSidePropsContext } from "next";
 import DashboardLayout from "../../components/layout/DashboardLayout";
 import { Common } from "../../constants/common";
 import { parseCookies } from "nookies";
+import {Calendar,Event,dateFnsLocalizer,momentLocalizer} from "react-big-calendar";
+import moment from "moment";
+
+const localizer = momentLocalizer(moment);
+
+import styles from "../../styles/pages/dashEvents.module.scss";
+import "react-big-calendar/lib/css/react-big-calendar.css"; // calendar css
+import { mockEvents } from "../../data/mockEvents";
+import EventPopup from "../../components/common/EventPopup/EventPopup";
+
+// see eventPropGetter
+const getEventClassByEvent = (event: Event) => {
+  let modifierStr = "";
+  if (event.resource[0]) {
+    const commitee = event.resource[0];
+    modifierStr = `rbc-override-${commitee}`;
+  }
+  return {
+    className: `rbc-override-event ${modifierStr}`,
+  };
+};
 
 function Events() {
-    return (
-        <DashboardLayout title="Events | ACM at PEC" heading="Events">
-            <p style={{ textAlign: "center" }}>Coming Soon! ✨</p>
-        </DashboardLayout>
-    );
+  const [listedEvents, setListedEvents] = useState<Event[] | undefined>( mockEvents);
+  const [activeEvent, setActiveEvent] = useState<Event | null>(null);
+  const [eventResource, setEventResource] = useState<string[]>(["acm","","","",]);
+  const [showModal, setShowModal] = useState<Boolean>(false);
+
+  const handleSelectEvent = (event: Event) => {
+    setEventResource(event ? event.resource : ["acm", "", "", ""]);
+    setActiveEvent(event);
+    setShowModal(true);
+  };
+
+  return (
+    <DashboardLayout title="Events | ACM at PEC" heading="Events">
+      <div className={styles.events}>
+        <p>Have a look at our calendar</p>
+        
+        <div className={styles.calendarWrapper}>
+        <div className={styles.eventButton}>
+          <button type="submit">Create Event</button>
+        </div>
```

Comment:
Make this part optional to see, only ExecutiveBody, Core and Admin role can see this one.

Role is stored in Cookie, in the `DESIGNATION` key, check Commons.ts to retrieve the cookie